subsys/bluetooth/host/id.c

```diff
@@ -773,12 +773,25 @@ void bt_id_add(struct bt_keys *keys)
 		goto done;
 	}
 
+	/*
+	 * Core Spec 5.3, p 2424
+	 * Controller can accept or reject when the same device is added to
+	 * resolving list.  This check ensures duplicate device will not be
+	 * sent to the controller.
+	 */
+	if (bt_dev.le.rl_entries) {
+		if (bt_keys_match(keys)) {
+			goto entry_added;
```

Comment:
can you do an if-else only ? there is already one goto, I don't want to complicate this fn more